library/src/lib/calendar/calendar-views/calendar2-year-view/calendar2-year-view.component.ts

```diff
@@ -1,17 +1,127 @@
+
+    /** @hidden */
+    ngAfterViewChecked() {
+        if (this.newFocusedYearId) {
+            this.focusElement(this.newFocusedYearId);
+            this.newFocusedYearId = null;
+        }
+    }
+
+    constructor(private eRef: ElementRef) { }
+
+    /** Method for handling the keyboard navigation */
+    onKeydownYearHandler(event, year: number, index: number) {
+        if (event.code === 'Tab' && !event.shiftKey) {
+            if (this.focusEscapeFunction) {
+                event.preventDefault();
+                this.focusEscapeFunction();
+            }
+        } else {
+            switch (event.code) {
+                case 'Enter':
+                case 'Space': {
+                    event.preventDefault();
+                    this.selectYear(year);
+                    break;
+                }
+                case 'ArrowLeft': {
+                    event.preventDefault();
+                    if (index === 0) {
+                        this.reGenerateYearList.emit('previousSet');
+                        this.newFocusedYearId = this.id + '-fd-year-' + 11;
+                    } else {
+                        this.newFocusedYearId = this.id + '-fd-year-' + (index - 1);
+                    }
+                    break;
+                }
+                case 'ArrowRight': {
+                    event.preventDefault();
+                    if (index === 11) {
+                        this.reGenerateYearList.emit('nextSet');
+                        this.newFocusedYearId = this.id + '-fd-year-' + 0;
+                    } else {
+                        this.newFocusedYearId = this.id + '-fd-year-' + (index + 1);
+                    }
+                    break;
+                }
+                case 'ArrowUp': {
+                    event.preventDefault();
+                    if (index <= 3) {
+                        this.reGenerateYearList.emit('previousSet');
+                        this.newFocusedYearId = this.id + '-fd-year-' + (index + 8);
+                    } else {
+                        this.newFocusedYearId = this.id + '-fd-year-' + (index - 4);
+                    }
+                    break;
+                }
+                case 'ArrowDown': {
+                    event.preventDefault();
+                    if (index >= 8) {
+                        this.reGenerateYearList.emit('nextSet');
+                        this.newFocusedYearId = this.id + '-fd-year-' + (index - 8);
+                    } else {
+                        this.newFocusedYearId = this.id + '-fd-year-' + (index + 4);
+                    }
+                    break;
+                }
+            }
+            this.focusElement(this.newFocusedYearId);
+        }
     }
 
-    ngOnInit() {
+    /** Method allowing focusing on elements within this component */
+    focusElement(elementSelector: string): void {
+        const elementToFocus: HTMLElement = this.eRef.nativeElement.querySelector('#' + elementSelector);
```

Comment:
I despise any use of querySelector in an Angular app, but since this is used in other views we can address the larger problem in the big PR which will follow this one.